DigitalLearningSolutions.Data/Models/EmailVerificationDetails.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace DigitalLearningSolutions.Data.Models
+{
+    using System;
+    using DigitalLearningSolutions.Data.Utilities;
+
+    public class EmailVerificationDetails
+    {
+        public int UserId { get; set; }
+        public string Email { get; set; } = null!;
+        public string EmailVerificationHash { get; set; } = null!;

```

Comment:
For my own clarification, why are you setting these to null and not make these nullable strings?

Reply:
`EmailVerificationHashes.EmailVerificationHash` is a not nullable database column. `GetCentreEmailVerificationDetails` (which is the only place this model is created) will always return either a `EmailVerificationDetails` (when in input code matches a value in `EmailVerificationHashes.EmailVerificationHash`), or null (when it doesn't). You can therefore guarantee that none of `UserId`, `Email`, `EmailVerificationHash`, or `EmailVerificationHashCreatedDate` will ever be null, because the model can only exist if they are not null. (`EmailVerified` is nullable because `UserCentreDetails.EmailVerified` is a nullable database column.)